### PR TITLE
chore(All plugins): Move all e2e tests to be integration_test

### PR DESCRIPTION
**Pull request by Chever-John**
create a new directory integration_test

## Description

Restructuring of files and directories in order for writing integration tests.

Additional Notes.
1. The integration test status of battery_plus is (**half done**) because the result of the integration test on the web page of this plugin shows that one function ( **isInBatterySaveMode()** ) is not implemented.
2. **Done for Android** means I have finished writing the integration test CI YAML file supporting the Android Platform.
3. **Done** under the **Android** column means we can run the command `flutter test integration_test/PLUGINS_NAME_test.dart` for our expected result.
4. **Done** under the **web** column means we can run the command `flutter drive --driver=test_driver/PLUGINS_NAME_test.dart --target=integration_test/PLUGIN_NAME_test.dart -d chrome` for our expected result.
5. One thing to note is that the **Android Alarm Manager** plugin does not get tested properly. I am looking for the cause of this.

| Plugin                   | Pub                            | Android | iOS | Web | MacOS | Linux | Windows | CI |
| ------------------------ | ------------------------------ | ------- | --- | --- | ----- | ----- | ------- | ------- |
| Battery ➕              | Battery ➕ Badge              | ✔     | ✔ | **Half Done** | ✔   | ✔   | ✔     | **Done for Android**     |
| Connectivity ➕         | Connectivity ➕ Badge         | ✔     | ✔ | **Done** | ✔   | ✔   | ✔     |  **Done for Android**     |
| Device Info ➕          | Device Info ➕ Badge          | ✔     | ✔ | **Done** | ✔   | ✔   | ✔     | **Done for Android**     |
| Network Info ➕         | Network Info ➕ Badge         | **Done**     | ✔ | ➖ | ✔   | ✔   | ✔     | **Done for Android**   |
| Package Info ➕         | Package Info ➕ Badge         | ✔     | ✔ | **Done** | ✔   | ✔   | ✔     | **Done for Android**     |
| Sensors ➕              | Sensors ➕ Badge              | **Done**     | ✔ | ✔ | ⨯   | ⨯   | ⨯     | **Done for Android**     |
| Share ➕                | Share ➕ Badge                | ✔     | ✔ | **Web** | ✔   | ✔   | ✔     | **Done for Android**     |
| Android Alarm Manager ➕ | Android Alarm Manager ➕ Badge | ✔     | ➖ | ➖ | ➖   | ➖   | ➖     | ✔     |
| Android Intent ➕       | Android Intent ➕ Badge       | **Done**     | ➖ | ➖ | ➖   | ➖   | ➖     | **Done**     |

## Related Issues

It should be related to https://github.com/fluttercommunity/plus_plugins/issues/958

#958

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by miquelbeltran** [comment]
I converted the PR as draft, you can set it to Ready for review when you are done ;)

**Comment by Chever-John** [comment]
@miquelbeltran 
yes, that's how it should be~

**Comment by miquelbeltran** [comment]
I saw that for share_plus, sensor_plus and  package_info_plus the tests didn't do anything, for example: https://github.com/fluttercommunity/plus_plugins/runs/7385366900?check_suite_focus=true

That is because the CI command does not use the same project name as the one in the examples folder.

You need to do some some changes, and verify that they are correct.

File: .github/workflows/device_info_plus.yaml [hunk 1]
```diff
@@ -24,9 +24,42 @@ jobs:
...
+      - name: "Bootstrap Workspace"
+        run: melos bootstrap
+      - name: "Android Integration Test"
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 30
+          arch: x86_64
+          target: google_apis
+          profile: Nexus 5X
+          script: ./.github/workflows/scripts/integration-test.sh android device_info_plus_example
```
Comment (miquelbeltran) [inline .github/workflows/device_info_plus.yaml:61]: should be `device_info_example`

File: .github/workflows/package_info_plus.yaml [hunk 2]
```diff
@@ -24,9 +24,42 @@ jobs:
...
+      - name: "Bootstrap Workspace"
+        run: melos bootstrap
+      - name: "Android Integration Test"
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 30
+          arch: x86_64
+          target: google_apis
+          profile: Nexus 5X
+          script: ./.github/workflows/scripts/integration-test.sh android package_info_plus_example
```
Comment (miquelbeltran) [inline .github/workflows/package_info_plus.yaml:61]: should be `package_info_example`

File: .github/workflows/sensors_plus.yaml [hunk 3]
```diff
@@ -24,9 +24,42 @@ jobs:
...
+      - name: "Bootstrap Workspace"
+        run: melos bootstrap
+      - name: "Android Integration Test"
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 30
+          arch: x86_64
+          target: google_apis
+          profile: Nexus 5X
+          script: ./.github/workflows/scripts/integration-test.sh android sensors_plus_example
```
Comment (miquelbeltran) [inline .github/workflows/sensors_plus.yaml:61]: should be `sensors_example`

File: .github/workflows/share_plus.yaml [hunk 4]
```diff
@@ -24,8 +24,41 @@ jobs:
...
+      - name: "Bootstrap Workspace"
+        run: melos bootstrap
+      - name: "Android Integration Test"
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 30
+          arch: x86_64
+          target: google_apis
+          profile: Nexus 5X
+          script: ./.github/workflows/scripts/integration-test.sh android share_plus_example
```
Comment (miquelbeltran) [inline .github/workflows/share_plus.yaml:61]: should be `share_example`

**Review by miquelbeltran — CHANGES_REQUESTED** [review]

File: .github/workflows/device_info_plus.yaml (hunk 1 above)
Comment (Chever-John) [reply .github/workflows/device_info_plus.yaml:61]: Copy that!

**Pull request merged**